src/engine/objects/resolvers/visitors.rs

```diff
@@ -986,13 +1089,26 @@ where
 
         let rel_label =
             rel_var.src().label()?.to_string() + &rel_var.label().to_title_case() + "Rel";
-        let rels = transaction
-            .read_rels::<RequestCtx>(fragment, rel_var, None, partition_key_opt)
+        let mut rels = transaction
+            .read_rels(fragment, rel_var, None, partition_key_opt)
             .await?;
         if rels.is_empty() {
             if let Some(handlers) = context.event_handlers().after_rel_delete(&rel_label) {
-                handlers.iter().try_fold(Vec::new(), |v, f| f(v))?;
-            }
+                let mut v = Vec::new();
+                for f in handlers.iter() {
+                    v = f(
+                        v,
+                        EventFacade::new(
+                            // TODO: where do we get rel_name ?
+                            CrudOperation::DeleteRel(info.name().to_string(), "?".to_string()),
```

Comment:
The RelQueryVar struct has a function to obtain it, so it'd likely be rel_var.label().

Reply:
Fixed.